### ISSUE TITLE
Proj compatibility with CMake on Ubuntu 20.04

### ISSUE THREAD

**Issue by lauralindzey** [OSGeo/PROJ#2744]
### Problem description

While the documentation suggests that [cmake + proj should be fine](https://proj.org/development/cmake.html), I have not yet been able to get this to work on Ubuntu 20.04. What I've tried:
* the apt-installed version doesn't install the cmake configuration files
* I can't compile from source according to the instructions because `cmake ..` fails
* I tried to find and patch in a PROJ4Config.cmake file. I haven't found one that works, and I'm not familiar enough with cmake to roll my own.

I've included more information about each attempt below; please let me know if additional details would be helpful for debugging.

#### Take 1: Just use apt

Unfortunately, the version in apt does not have the cmake configure files
correctly installed. This appears to be a known issue:
* https://github.com/OSGeo/PROJ/issues/1484
* https://github.com/OSGeo/PROJ/issues/2546

Additionally, it is rather old (6.3.1), which made me less inclined to use it.

#### Take 2: Try installing from source?

Next, I tried to [install manually from source using cmake](https://proj.org/install.html#cmake), which was a suggested workaround:
~~~
git clone https://github.com/OSGeo/PROJ.git
cd PROJ
mkdir build; cd build
cmake ..
~~~

Unfortunately, this install is failing for me. I have tried with master, 8.0.1 and 7.2;
they all have the same errors.

I *think* that the relevant line there is `/usr/bin/ld: cannot find -lpthreads`

However, I seem to have pthread installed on my system:

~~~bash
llindzey@llindzey-linux-1:~/PROJ/build$ find /usr/lib -iname "*pthread*"
/usr/lib/x86_64-linux-gnu/libpthread.a
/usr/lib/x86_64-linux-gnu/libpthread.so.0
/usr/lib/x86_64-linux-gnu/libevent_pthreads-2.1.so.7.0.0
/usr/lib/x86_64-linux-gnu/libgpgme-pthread.so
/usr/lib/x86_64-linux-gnu/libpthread.so
/usr/lib/x86_64-linux-gnu/libgpgme-pthread.so.11
/usr/lib/x86_64-linux-gnu/libevent_pthreads.so
/usr/lib/x86_64-linux-gnu/libevent_pthreads.a
/usr/lib/x86_64-linux-gnu/pkgconfig/pthread-stubs.pc
/usr/lib/x86_64-linux-gnu/pkgconfig/libevent_pthreads.pc
/usr/lib/x86_64-linux-gnu/perl/5.30.0/bits/pthreadtypes-arch.ph
/usr/lib/x86_64-linux-gnu/perl/5.30.0/bits/pthreadtypes.ph
/usr/lib/x86_64-linux-gnu/libpthread-2.31.so
/usr/lib/x86_64-linux-gnu/libevent_pthreads-2.1.so.7
~~~

Any idea what the problem could be?

#### Take 3: Create my own PROJ4Config.cmake file

As I understand it, one possible fix is to version control my own PROJ4Config.cmake file, and point cmake at it using:
~~~
find_package(PROJ4 REQUIRED PATHS ${CMAKE_CURRENT_LIST_DIR}/../../cmake)
~~~

I *think* that an approach like this should work, and I've figured out how to get cmake to find the file.
However, I'm not sure what belongs in that file, and simply copying [this one](https://github.com/OSGeo/PROJ/blob/master/cmake/ProjConfig.cmake) from the repo did not succeed. ($PROJ4_LIBRARIES and $PROJ4_INCLUDE_DIRS are both empty, when checked by `message(STATUS ...)`)

However, while I can compile, I start getting linker errors.

Does anybody have a cmake config file that works with the version of proj installed by apt (6.3.1)?

**Comment by rouault** [comment]
regarding build issues on Ubuntu 20.04, this is really surprising. This is my dev environment. You should be more specific and attach the output of cmake & make

a successfull build can be done with:
```
docker run --rm -it ubuntu:20.04
# within the container
apt update
apt install git cmake make g++ libtiff-dev sqlite3 libsqlite3-dev libcurl4-gnutls-dev
git clone https://github.com/OSGeo/PROJ.git
cd PROJ/
mkdir build
cd build/
cmake ..
make -j$(nproc)
```

**Comment by lauralindzey** [comment]
OK, thanks -- it looks like I had the wrong version of sqlite installed. So, I can now build from source. My apologies.

Is it an accurate summary of cmake + proj to say that:
* If you want to use the version in apt, don't bother with trying to get `find_project` to work and instead just explicitly add the path to the header files + the library as above 
* If you want `find_project(PROJ)` to work, install from source.
?

**Comment by rouault** [comment]
> Is it an accurate summary of cmake + proj to say that:

my memory is notoriously bad, but there have been various fixes done after proj 6.3.2 regarding the cmake integration, so yes you might have issues with find_project in those older versions.

**Comment by mwtoews** [comment]
Ubuntu (and Debian) use Autotools for packaging and distributing PROJ via apt, so it does not contain the necessary CMake files to enable find_project to work as expected. One CMake work-around is to use [FindPkgConfig](https://cmake.org/cmake/help/latest/module/FindPkgConfig.html), which I've not yet tested.

On a related note, I'm slowly putting together an RFC to drop Autotools and support only CMake.
